From 0ecd9a320a94998f1b7066b5997b3e43a9c8a586 Mon Sep 17 00:00:00 2001
From: Thiago Macieira <thiago.macieira@intel.com>
Date: Tue, 7 May 2019 16:04:28 -0700
Subject: [PATCH] Make the non-X11 dbus-launch exec the X11-enabled one

This allows us to have dbus-launch without the X11 dependency, in separate
packages.

Signed-off-by: Thiago Macieira <thiago.macieira@intel.com>
---
 tools/dbus-launch.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/tools/dbus-launch.c b/tools/dbus-launch.c
index f1a5b970..89360bea 100644
--- a/tools/dbus-launch.c
+++ b/tools/dbus-launch.c
@@ -1007,6 +1007,15 @@ main (int argc, char **argv)
   if (exit_with_x11)
     verbose ("--exit-with-x11 enabled\n");
 
+#ifndef DBUS_BUILD_X11
+  /* Attempt to re-exec the X11 version */
+  if (getenv("DISPLAY"))
+    {
+      (void) execv (DBUS_BINDIR "/dbus-launch.x11", argv);
+      /* ignore any failures, we'll print errors below if necessary */
+    }
+#endif
+
   if (autolaunch)
     {      
 #ifndef DBUS_BUILD_X11
-- 
2.21.0

